### DISCUSSION TITLE
Captive runtime build failing when I'm adding InfoAdditions

### DISCUSSION THREAD

**Discussion by piotrzarzycki21** [airsdk/Adobe-Runtime-Support#2961] [General]
I'm not sure what I'm doing wrong in that regards. I have following part of code (ant script) which package application to captive runtime:

```
<java jar="${ADT_PATH}" fork="true" failonerror="true">
			<arg line="-package" />
			<arg line="-storetype pkcs12" />
			<arg line="-keystore '${MAC_APPLICATION_CERTKEY}' " />
			<arg line="-storepass '${MAC_APPLICATION_CERTKEY_PASS}'" />
			<arg line="-tsa 'http://timestamp.digicert.com'" />
			<arg line="-target bundle"/>
			<arg line="'${DEPLOY_DIR}/${APP_PACKAGE_NAME}'" />
			<arg line="${DESKTOP_DESCRIPTOR}" />
			<arg line="${SWF_FILE_PATH}" />
			<arg line="assets" />
</java>
```
I would like to add something to Info.plist file apart what has been already added. If I understand correctly in application descriptor file I can add this part:

```
<macOS> 
        <InfoAdditions>
            <![CDATA[
            		<key>CFBundleName</key>
                <string>Some value</string>
            ]]>
        </InfoAdditions> 
     </macOS>
```

When I added it I'm getting following error:
`Main-app.xml(125): error 105: application.macOS.InfoAdditions contains an invalid value`

This is desktop application.
**Mac OS: Sonoma 14.1.2 (23B92)
Harman Air: 50.2.3.8**

**Comment by ajwfrost** [comment]
Hi

It looks like `CFBundleName` was added to the reserved list of keys, back in AIR 3.1 (albeit of course just for IPA files at that point..) -> but the value is populated automatically as part of the mac bundle generation, using the `name` field from your application descriptor file. So it shouldn't be necessary to set it via InfoAdditions, you can just set it directly as your app name..?

thanks

**Comment by piotrzarzycki21** [comment]
> It looks like `CFBundleName` was added to the reserved list of keys, back in AIR 3.1 (albeit of course just for IPA files at that point..)

Where can I check list of restricted properties in that matter ?

**Comment by piotrzarzycki21** [comment]
> So it shouldn't be necessary to set it via InfoAdditions, you can just set it directly as your app name..?

The thing is that my app name (CFBundleExecutable) is different than CFBundleName - This causes an issue where In application menu I have that app name displayed instead CFBundleName. I was trying to test trough InfoAdditions some options which allows me to change that. 

<img width="171" alt="Screenshot 2023-12-12 at 12 28 59" src="https://github.com/airsdk/Adobe-Runtime-Support/assets/24554795/4b2d4cb8-8dba-48be-afae-11c57e6d182e">

Now I understand that some stuff is restricted.

**Comment by ajwfrost** [comment]
Hmm... so `CFBundleExecutable` seems to come from the `filename` field in the app descriptor file. So you should be able to set those differently:
```
	<filename>Appears_in_CFBundleExecutable</filename>
	<name>Appears_in_CFBundleName</name>
```

Banned tags list:
```
CFBundleVersion
CFBundleShortVersionString
CFBundleIdentifier
CFBundleInfoDictionaryVersion
CFBundleExecutable
CFBundleDisplayName
CFBundlePackageType
CFBundleIconFiles
CFBundleSupportedPlatforms
CFBundleResourceSpecification
CTAutoOrients
CTInitialWindowVisible
CTInitialWindowTitle
CTNamespaceURI
DTPlatformName
DTSDKName
DTSDKBuild
DTCompiler
DTPlatformBuild
DTPlatformName
DTPlatformVersion
DTXcode
DTXcodeBuild
LSRequiresIPhoneOS
NSMainNibFile
UIStatusBarHidden
UIInterfaceOrientation
UISupportedInterfaceOrientations
CFBundleName
```

Although see also #2857 - I don't actually see a particular reason why it shouldn't be allowed to override some of these sorts of things if people want to!

**Comment by piotrzarzycki21** [comment]
Thanks @ajwfrost. It's more clear now. It just looks like it may not be possible to have some custom text in About section in menu - maybe it's always aligned with executable name.
